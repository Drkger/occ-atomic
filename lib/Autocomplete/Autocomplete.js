import React from 'react';
import PropTypes from 'prop-types';

import Input from '../Input';
import Droplist from '../Droplist';
import Label from '../Label';

/**
* This component integrates the Input, Droplist and Label components.
* The value of Input filters the list in Droplist.
* The selected items of Droplist can be stacked into Labels.
*/
class Autocomplete extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            value: props.inputProps.valueProp ? props.inputProps.valueProp : '',
            focus: false,
            labels: []
        };
        this.onChange = this.onChange.bind(this);
        this.onFocus = this.onFocus.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onKeyUp = this.onKeyUp.bind(this);
        this.onClear = this.onClear.bind(this);
        this.onClick = this.onClick.bind(this);
        this.addLabel = this.addLabel.bind(this);
        this.removeLabel = this.removeLabel.bind(this);
        this.removeSelected = this.removeSelected.bind(this);
        this.forceValue = this.forceValue.bind(this);
        this.setRef = this.setRef.bind(this);
        this.labels = [];
    }

    componentWillMount() {
        let { onRef } = this.props;
        if (onRef) onRef(this);
    }

    componentWillUnmount() {
        let { onRef } = this.props;
        if (onRef) onRef(undefined);
    }

    onChange(value) {
        const { onChange } = this.props;
        this.setState({
            value, focus: true
        });
        if (onChange) onChange(value);
    }

    onFocus() {
        const { onFocus } = this.props;
        this.setState({
            focus: true
        });
        if (onFocus) onFocus();
    }

    onBlur() {
        const { onBlur } = this.props;
        setTimeout(() => {
            this.setState({
                focus: false
            });
        }, 200);
        if (onBlur) onBlur();
    }

    onKeyUp(key) {
        const { onKeyUp, selectOnEnter, droplistProps } = this.props;
        if (selectOnEnter) {
            if (key == 13) {
                const cleanItems = this.removeSelected();
                if (cleanItems.length) {
                    if (!droplistProps.groups) {
                        const item = cleanItems[0];
                        this.onClick(item);
                    } else {
                        const item = cleanItems[0][droplistProps.groupItemsKey][0];
                        this.onClick(item);
                    }
                } else {
                    this.setState({
                        value: ''
                    });
                }
            }
        }
        if (onKeyUp) onKeyUp(key);
    }

    onClear() {
        const { onClear } = this.props;
        this.setState({
            value: ''
        });
        if (onClear) onClear();
    }

    onClick(item) {
        const { onClick, onChange, droplistProps, stackLabels } = this.props;
        if (stackLabels) {
            this.input.onClear();
            this.setState({
                value: ''
            });
            this.input.input.focus();
            this.addLabel(item);
        } else {
            this.setState({
                value: item[droplistProps.itemTextKey], focus: false
            });
            if (onChange) onChange(item[droplistProps.itemTextKey]);
        }
        if (onClick) onClick(stackLabels ? this.labels : item);
    }

    addLabel(item) {
        this.labels.push(item);
        this.setState({labels: this.labels});
    }

    removeLabel(item) {
        const { droplistProps, onClick } = this.props;
        let pos = this.labels.map(label => { return label[droplistProps.itemIdKey];}).indexOf(item[droplistProps.itemIdKey]);
        this.labels.splice(pos, 1);
        this.setState({labels: this.labels});
        if (onClick) onClick(this.labels);
    }

    removeSelected() {
        const { labels } = this.state;
        const { droplistProps } = this.props;
        let cleanItems = [];
        if (droplistProps.groups) {
            droplistProps.items.map(group => {
                let cleanGroup = group[droplistProps.groupItemsKey].filter(item => {
                    const inLabels = labels.filter(label => {
                        return label[droplistProps.itemIdKey] == item[droplistProps.itemIdKey];
                    });
                    return !inLabels.length;
                });
                const modifiedGroup = Object.assign({}, group);
                modifiedGroup[droplistProps.groupItemsKey] = cleanGroup;
                if (cleanGroup.length) cleanItems.push(modifiedGroup);
            });
        } else {
            cleanItems = droplistProps.items.filter(item => {
                const inLabels = labels.filter(label => {
                    return label[droplistProps.itemIdKey] == item[droplistProps.itemIdKey];
                });
                return !inLabels.length;
            });
        }
        return cleanItems;
    }

    forceValue(value) {
        this.setState({value});
    }

    setRef(input) {
        if (input)
            this.input = input;
    }

    render() {
        const { value, focus, labels } = this.state;
        const { classes, id, inputProps, droplistProps, stackLabels, limit, limitPlaceholder, className, style, disableAutoComplete } = this.props;
        const cleanItems = this.removeSelected();
        return (
            <div className={`${classes.autoComplete}${className ? ` ${className}` : ''}`} id={id} style={style} ref = {ref => { this.autocomplete = ref; }}>
                <Input
                    onRef={this.setRef}
                    id={inputProps.id}
                    type={inputProps.type}
                    name={inputProps.name}
                    label={inputProps.label}
                    placeholder={((stackLabels && limit) && this.labels.length >= limit) ? limitPlaceholder : inputProps.placeholder}
                    onChange={this.onChange}
                    onFocus={this.onFocus}
                    onBlur={this.onBlur}
                    onKeyUp={this.onKeyUp}
                    onClear={this.onClear}
                    selectOnFocus={inputProps.selectOnFocus}
                    clear={inputProps.clear}
                    disabled={(stackLabels && limit) && this.labels.length >= limit}
                    valueProp={value}
                    iconName={inputProps.iconName}
                    theme={inputProps.theme}
                    className={inputProps.className}
                    inputClassName={inputProps.inputClassName}
                    disableAutoComplete={disableAutoComplete}
                />
                {(value && focus && (cleanItems.length)) ? (
                    <Droplist
                        items={cleanItems}
                        groups={droplistProps.groups}
                        groupNameKey={droplistProps.groupNameKey}
                        groupIdKey={droplistProps.groupIdKey}
                        groupItemsKey={droplistProps.groupItemsKey}
                        itemTextKey={droplistProps.itemTextKey}
                        itemTextRightKey={droplistProps.itemTextRightKey}
                        itemIdKey={droplistProps.itemIdKey}
                        term={value}
                        filter={droplistProps.filter}
                        onClick={this.onClick}
                        className={`${classes.droplist}${droplistProps.className ? ` ${droplistProps.className}` : ''}`} />
                ) : null}
                {(stackLabels && labels.length) ? (
                    <div className={classes.labels}>
                        {
                            labels.map(label => (
                                <Label key={label[droplistProps.itemIdKey]} onClose={() => {this.removeLabel(label);}}>{label[droplistProps.itemTextKey]}</Label>
                            ))
                        }
                    </div>
                ) : ''}
            </div>
        );
    }
}

Autocomplete.propTypes = {
    classes: PropTypes.object,
    /** Object with the props for the Input component. Check their descriptions in the Input docs. */
    inputProps: PropTypes.shape({
        id: PropTypes.string,
        type: PropTypes.string,
        name: PropTypes.string,
        label: PropTypes.string,
        placeholder: PropTypes.string,
        selectOnFocus: PropTypes.bool,
        clear: PropTypes.bool,
        iconName: PropTypes.string,
        theme: PropTypes.string,
        valueProp: PropTypes.string,
        className: PropTypes.string,
        inputClassName: PropTypes.string
    }).isRequired,
    /** Object with the props for the Droplist component. Check their descriptions in the Droplist docs. */
    droplistProps: PropTypes.shape({
        items: PropTypes.array,
        itemIdKey: PropTypes.string,
        itemTextKey: PropTypes.string,
        itemTextRightKey: PropTypes.string,
        groups: PropTypes.bool,
        groupIdKey: PropTypes.string,
        groupNameKey: PropTypes.string,
        groupItemsKey: PropTypes.string,
        className: PropTypes.string,
        filter: PropTypes.bool
    }).isRequired,
    /** Function to call on change (Input). */
    onChange: PropTypes.func,
    /** Function to call on key up (Input). */
    onKeyUp: PropTypes.func,
    /** Function to call on focus (Input). */
    onFocus: PropTypes.func,
    /** Function to call on blur (Input). */
    onBlur: PropTypes.func,
    /** Function to call on clear (Input). */
    onClear: PropTypes.func,
    /** Function to call on click (Droplist item). */
    onClick: PropTypes.func,
    onRef: PropTypes.func,
    /** Set this property to stack the selected items of Droplist into Labels. */
    stackLabels: PropTypes.bool,
    /** Set the limit of stackable items into labels. */
    limit: PropTypes.number,
    /** Change the placeholder when the limit of labels has been reached. */
    limitPlaceholder: PropTypes.string,
    /** Automatically selects the first item of the Droplist when pressing the Enter key. NOTE: To use this property correctly you must set the droplistProps.filter property as false and handle the filtering yourself, since Autocomplete only has access to the droplistProps.items property and not the items already filtered by Droplist. */
    selectOnEnter: PropTypes.bool,
    /**  Disables native autoComplete on browsers. */
    disableAutoComplete: PropTypes.bool,
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object
};

Autocomplete.defaultProps = {
    inputProps: {},
    droplistProps: {},
    disableAutoComplete: false
};

export default Autocomplete;
