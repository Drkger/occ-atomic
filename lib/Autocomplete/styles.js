import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
