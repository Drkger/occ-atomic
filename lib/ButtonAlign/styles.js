export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};