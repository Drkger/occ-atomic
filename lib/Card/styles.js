import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;