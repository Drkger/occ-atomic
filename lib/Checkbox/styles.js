import colors from '../subatomic/colors';
import spacing from '../subatomic/spacing';
import icons from '../subatomic/icons';

export default {
    cont: {
        paddingTop:spacing.tiny,
        paddingBottom:spacing.tiny,
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        cursor:'pointer',
        outline:'0',
        '&:after': {
            content:'""',
            display:'table',
            clear:'both'
        },
        '&:hover $check:after': {
            background:icons.base(icons.check.icon([colors.grey200]))
        }
    },
    check: {
        width:spacing.base,
        height:spacing.base,
        position:'relative',
        '&:before': {
            content:'""',
            width:spacing.small,
            height:spacing.small,
            borderRadius:4,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            border:`1px solid ${colors.grey200}`,
            background:colors.bgWhite
        },
        '&:after': {
            content:'""',
            width:18,
            height:18,
            marginTop:-1,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            transition:'0.3s all',
            background:'transparent'
        }
    },
    active: {
        '& $check:after': {
            background:[icons.base(icons.check.icon([colors.prim])), '!important']
        }
    },
    disabled: {
        opacity:0.4,
        pointerEvents:'none'
    },
    label: {
        marginLeft:spacing.tiny,
        cursor:'pointer'
    }
};
