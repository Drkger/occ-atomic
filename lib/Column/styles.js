import grid from '../subatomic/grid';

const styles = {
    col: {
        boxSizing:'border-box',
        position:'relative',
        minHeight:'1px',
        paddingRight:'15px',
        paddingLeft:'15px'
    }
};

const pushGrid = function() {
    const sm = `@media (min-width: ${grid.sm}px)`;
    const md = `@media (min-width: ${grid.md}px)`;
    const lg = `@media (min-width: ${grid.lg}px)`;
    for (let i = 1; i <= 12; i++) {
        styles[`xs${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[`offset-xs${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[`push-xs${i}`] = {
            left:`${100/12*i}%`
        };
        styles[`pull-xs${i}`] = {
            right:`${100/12*i}%`
        };
    };
    styles[sm] = {};
    styles[md] = {};
    styles[lg] = {};
    for(let i = 1; i <= 12; i++) {
        styles[sm][`sm${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[sm][`offset-sm${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[sm][`push-sm${i}`] = {
            left:`${100/12*i}%`
        };
        styles[sm][`pull-sm${i}`] = {
            right:`${100/12*i}%`
        };
        styles[md][`md${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[md][`offset-md${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[md][`push-md${i}`] = {
            left:`${100/12*i}%`
        };
        styles[md][`pull-md${i}`] = {
            right:`${100/12*i}%`
        };
        styles[lg][`lg${i}`] = {
            float:'left',
            width:`${100/12*i}%`
        };
        styles[lg][`offset-lg${i}`] = {
            marginLeft:`${100/12*i}%`
        };
        styles[lg][`push-lg${i}`] = {
            left:`${100/12*i}%`
        };
        styles[lg][`pull-lg${i}`] = {
            right:`${100/12*i}%`
        };
    }
};

pushGrid();

export default styles;