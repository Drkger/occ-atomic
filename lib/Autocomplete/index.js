import injectSheet from 'react-jss';
import Autocomplete from './Autocomplete';
import styles from './styles';

export default injectSheet(styles)(Autocomplete);