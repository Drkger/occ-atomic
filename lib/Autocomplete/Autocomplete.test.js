import React from 'react';
import { shallow, mount } from 'enzyme';
import Autocomplete from './Autocomplete';
import styles from './styles';
import AutocompleteJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

const items = [
    { text: 'Administrador', textRight: '(222)', id:1 },
    { text: 'Administrador Sr', textRight: '(134)', id:2 },
    { text: 'Administrador Jr', textRight: '(52)', id:3 },
    { text: 'Administrar', textRight: '(6)', id:4 },
];

// const items2 = [
//     { text: 'Países', id:'1',
//         items: [
//             { text: 'México', textRight: '(3405)', key: 'MX', type: 'CO', id:'1.1' }
//         ]
//     },
//     { text: 'Estados', id:'2',
//         items: [
//             { text: 'Querétaro', textRight: '(222)', key: 'QRO', type: 'ST', id:'2.1' },
//             { text: 'Estado de México', textRight: '(225)', key: 'EMX', type: 'ST', id:'2.2' }
//         ]
//     },
//     { text: 'Ciudades', id:'3',
//         items: [
//             { text: 'Santiago de Querétaro', textRight: '(102)', key: 'QRO2', type: 'CY', id:'3.1' }
//         ]
//     }    
// ];

describe("Autocomplete", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Autocomplete droplistProps={{items}} classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the input and droplist', () => {
        const wrapper = mount(
            <Autocomplete inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.setState({focus: true});
        expect(wrapper.find('.input').at(0).length).toBe(1);
        expect(wrapper.find('.droplist').at(0).length).toBe(1);
    });

    it('hides the droplist when there\'s no value or the input has no focus', () => {
        const wrapper = mount(
            <Autocomplete inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.setState({focus: true});
        expect(wrapper.find('.droplist').at(0).length).toBe(1);
        wrapper.setState({focus: false});
        expect(wrapper.find('.droplist').at(0).length).toBe(0);
        wrapper.setState({focus: true, value:''});
        expect(wrapper.find('.droplist').at(0).length).toBe(0);
    });

    it('filters the items of the droplist using the input value', () => {
        const wrapper = mount(
            <Autocomplete inputProps={{className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.setState({focus: true, value:'a'});
        expect(wrapper.find('.droplist').at(0).text()).toBe('Administrador(222)Administrador Sr(134)Administrador Jr(52)Administrar(6)');
        wrapper.setState({value:'administrador'});
        expect(wrapper.find('.droplist').at(0).text()).toBe('Administrador(222)Administrador Sr(134)Administrador Jr(52)');
    });
    
    // it('calls the customFilter function', () => {
    // 
    // });

    it('returns the input value', () => {
        const onChange = jest.fn();
        const wrapper = mount(
            <Autocomplete onChange={onChange} inputProps={{className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.find('input').simulate('change', {target: {value:'a'}});
        expect(onChange.mock.calls[0][0]).toBe('a');
        wrapper.find('input').simulate('change', {target: {value:'ad'}});
        expect(onChange.mock.calls[1][0]).toBe('ad');
    });

    it('returns the object of the selected item', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <Autocomplete onClick={onClick} inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.setState({focus: true});
        wrapper.find('.Droplist-item-0-1-48').at(0).simulate('click', items[0]);
        expect(onClick.mock.calls[0][0]).toEqual({id: 1, text: 'Administrador', textRight: '(222)'});
        wrapper.setState({focus: true, value:'a'});
        wrapper.find('.Droplist-item-0-1-48').at(1).simulate('click', items[0]);
        expect(onClick.mock.calls[1][0]).toEqual({id: 2, text: 'Administrador Sr', textRight: '(134)'});
    });

    it('stacks the selected items into labels', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <Autocomplete onClick={onClick} stackLabels inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist', itemIdKey:'id'}} classes={classes} />
        );
        wrapper.setState({focus: true});
        wrapper.find('.Droplist-item-0-1-48').at(0).simulate('click');
        expect(onClick.mock.calls[0][0]).toEqual([{id: 1, text: 'Administrador', textRight: '(222)'}]);
        wrapper.setState({focus: true, value:'ad'});
        wrapper.find('.Droplist-item-0-1-48').at(0).simulate('click');
        expect(onClick.mock.calls[1][0]).toEqual([{"id": 1, "text": "Administrador", "textRight": "(222)"}, {"id": 2, "text": "Administrador Sr", "textRight": "(134)"}]);
    });

    it('disables the input when it reaches the limit of labels', () => {
        const wrapper = mount(
            <Autocomplete stackLabels limit={2} inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist', itemIdKey:'id'}} classes={classes} />
        );
        wrapper.setState({focus: true});
        expect(wrapper.find('input').length).toBe(1);
        wrapper.find('.Droplist-item-0-1-48').at(0).simulate('click');
        wrapper.setState({focus: true, value:'ad'});
        expect(wrapper.find('input').length).toBe(1);
        wrapper.find('.Droplist-item-0-1-48').at(0).simulate('click');
        wrapper.setState({focus: true, value:'ad'});
        expect(wrapper.find('input').length).toBe(0);
    });

    it('deletes a label', () => {
        const wrapper = mount(
            <Autocomplete stackLabels inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist', itemIdKey:'id'}} classes={classes} />
        );
        wrapper.setState({focus: true});
        expect(wrapper.state('labels')).toEqual([]);
        expect(wrapper.find('label').length).toBe(0);
        wrapper.find('.Droplist-item-0-1-48').at(0).simulate('click');
        expect(wrapper.state('labels')).toEqual([{"id": 1, "text": "Administrador", "textRight": "(222)"}]);
        expect(wrapper.find('label').length).toBe(1);
        wrapper.find('label span').simulate('click');
        expect(wrapper.state('labels')).toEqual([]);
        expect(wrapper.find('label').length).toBe(0);
    });

    it('calls the onChange function', () => {
        const onChange = jest.fn();
        const wrapper = mount(
            <Autocomplete onChange={onChange} inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.find('input').simulate('change', {target: {value:'a'}});
        expect(onChange.mock.calls.length).toBe(1);
    });

    it('calls the onClick function', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <Autocomplete onClick={onClick} inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.setState({focus: true});
        console.log(wrapper.html());
        wrapper.find('.Droplist-item-0-1-48').at(0).simulate('click');
        expect(onClick.mock.calls.length).toBe(1);
    });

    it('calls the onFocus function', () => {
        const onFocus = jest.fn();
        const wrapper = mount(
            <Autocomplete onFocus={onFocus} inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.find('input').simulate('focus');
        expect(onFocus.mock.calls.length).toBe(1);
    });

    it('calls the onBlur function', () => {
        const onBlur = jest.fn();
        const wrapper = mount(
            <Autocomplete onBlur={onBlur} inputProps={{valueProp:'a', className:'input'}} droplistProps={{items, className:'droplist'}} classes={classes} />
        );
        wrapper.find('input').simulate('blur');
        expect(onBlur.mock.calls.length).toBe(1);
    });
});

describe("AutocompleteJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <AutocompleteJSS droplistProps={{items}} items={items} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Autocomplete styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});