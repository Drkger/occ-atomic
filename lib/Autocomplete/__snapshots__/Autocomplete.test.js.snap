// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Autocomplete
    classes={
      Object {
        "autoComplete": "autoComplete",
        "droplist": "droplist",
        "labels": "labels",
      }
    }
    droplistProps={
      Object {
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      }
    }
    inputProps={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StyledInput
          className="undefined"
          clear={undefined}
          disabled={undefined}
          iconName={undefined}
          id={undefined}
          label={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClear={[Function]}
          onFocus={[Function]}
          onKeyUp={[Function]}
          onRef={[Function]}
          placeholder={undefined}
          selectOnFocus={undefined}
          theme={undefined}
          type={undefined}
          valueProp=""
        />,
        null,
        "",
      ],
      "className": "autoComplete",
      "id": undefined,
      "style": undefined,
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "undefined",
          "clear": undefined,
          "disabled": undefined,
          "iconName": undefined,
          "id": undefined,
          "label": undefined,
          "name": undefined,
          "onBlur": [Function],
          "onChange": [Function],
          "onClear": [Function],
          "onFocus": [Function],
          "onKeyUp": [Function],
          "onRef": [Function],
          "placeholder": undefined,
          "selectOnFocus": undefined,
          "theme": undefined,
          "type": undefined,
          "valueProp": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StyledInput
            className="undefined"
            clear={undefined}
            disabled={undefined}
            iconName={undefined}
            id={undefined}
            label={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            onRef={[Function]}
            placeholder={undefined}
            selectOnFocus={undefined}
            theme={undefined}
            type={undefined}
            valueProp=""
          />,
          null,
          "",
        ],
        "className": "autoComplete",
        "id": undefined,
        "style": undefined,
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "undefined",
            "clear": undefined,
            "disabled": undefined,
            "iconName": undefined,
            "id": undefined,
            "label": undefined,
            "name": undefined,
            "onBlur": [Function],
            "onChange": [Function],
            "onClear": [Function],
            "onFocus": [Function],
            "onKeyUp": [Function],
            "onRef": [Function],
            "placeholder": undefined,
            "selectOnFocus": undefined,
            "theme": undefined,
            "type": undefined,
            "valueProp": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Autocomplete styles matches the snapshot 1`] = `
Object {
  "autoComplete": Object {
    "position": "relative",
  },
  "droplist": Object {
    "left": "0",
    "position": "absolute",
    "top": [Function],
    "width": "100%",
    "zIndex": "2",
  },
  "labels": Object {
    "marginTop": "-18px",
    "width": "100%",
  },
}
`;

exports[`AutocompleteJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Autocomplete)
    droplistProps={
      Object {
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      }
    }
    inputProps={Object {}}
    items={
      Array [
        Object {
          "id": 1,
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "id": 2,
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "id": 3,
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "id": 4,
          "text": "Administrar",
          "textRight": "(6)",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "autoComplete": "Autocomplete-autoComplete-0-1-693 Autocomplete-autoComplete-0-1-689",
        "droplist": "Autocomplete-droplist-0-1-692 Autocomplete-droplist-0-1-690",
        "labels": "Autocomplete-labels-0-1-694 Autocomplete-labels-0-1-691",
      },
      "droplistProps": Object {
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      },
      "inputProps": Object {},
      "items": Array [
        Object {
          "id": 1,
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "id": 2,
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "id": 3,
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "id": 4,
          "text": "Administrar",
          "textRight": "(6)",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "autoComplete": "Autocomplete-autoComplete-0-1-693 Autocomplete-autoComplete-0-1-689",
          "droplist": "Autocomplete-droplist-0-1-692 Autocomplete-droplist-0-1-690",
          "labels": "Autocomplete-labels-0-1-694 Autocomplete-labels-0-1-691",
        },
        "droplistProps": Object {
          "items": Array [
            Object {
              "id": 1,
              "text": "Administrador",
              "textRight": "(222)",
            },
            Object {
              "id": 2,
              "text": "Administrador Sr",
              "textRight": "(134)",
            },
            Object {
              "id": 3,
              "text": "Administrador Jr",
              "textRight": "(52)",
            },
            Object {
              "id": 4,
              "text": "Administrar",
              "textRight": "(6)",
            },
          ],
        },
        "inputProps": Object {},
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
